### DISCUSSION TITLE
Autorun-rs v1.1.0

### DISCUSSION THREAD

**Discussion by vurvdev** [vurvdev/Autorun-rs#48] [Announcements]
A lot of not very minor but also not too crazy changes were made in 1.1.0

### ``Autorun.require``
This will now pass errors encountered in the function being ran, if any, instead of failing silently.

## New Logger 📋
I replaced the very ugly logger with a custom implementation, which saves them in ``YYYY-MM-DD`` format for file sorting, and logs them with colors to the external console. 
![image](https://user-images.githubusercontent.com/56230599/160051990-41f5f07c-52a0-43c8-bf5f-9a401472bb26.png)
![image](https://user-images.githubusercontent.com/56230599/160052070-8960da79-0698-471b-a488-45e1a768fcfa.png)

## Asynchronous Lua Dumping 🌔
When joining servers, lua will be dumped asynchronously on a different thread, dumping around 50 files a second max to avoid using too much of your disk. (There could also be a setting to change the cooldown, bug me if you need it)

## Console Enhancements ✨
The console finally has proper colors, a command system for easy extension and descriptions builtin.
Check it out:
![image](https://user-images.githubusercontent.com/56230599/160051348-50ecb6f9-c3e6-4638-a45e-5ab41388a0bd.png)

### New command: ``clear`` 🌀
This new command will clear the console, very useful when you're using something like [Safety](https://github.com/Vurv78/Safety) which prints a lot to the console..

### ``Autorun.requirebin(name: string)`` ⚙️ 
This function is used to execute binary modules (Rust, C++ programs) located in your ``autorun/bin`` folder, as long as they are named accordingly: e.g. ``gmcl_mymodule_win64.dll`` for ``Autorun.requirebin("mymodule")``

### ``Autorun.print(...)`` 💻
Accompanying the colors in the console, you can now print with color using the Autorun.print command, which accepts any type of arguments, including tables, which, if they contain a number as the first index, will be considered a color.

**Example Use:**
```lua
Autorun.print( {255, 0, 0}, "Hello world!", {0, 0, 255}, "Now in blue!!", {0, 255, 0}, "Now with a function!!!", print )
```
![image](https://user-images.githubusercontent.com/56230599/160054604-2fe96d1e-b44b-47f1-9e86-3ccd1183fb28.png)

## Once Again...

That's pretty much it. It took a while to get here, so if you could support with a 🌟 or [join the discord to ask any questions or report issues](https://discord.gg/yXKMt2XUXm), it would be greatly appreciated. (This took way too long to write, so could support me for that as well.. 😵‍💫 )

<hr /><em>This discussion was created from the release <a href='https://github.com/Vurv78/Autorun-rs/releases/tag/v1.1.0'>Autorun-rs v1.1.0</a>.</em>

**Comment by Varnek** [comment]
Since nobody is saying thank you, I will. Thank you for making a useful tool such as this and releasing this to the public, will definetly use it and maybe learn from it :)
